crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,1459 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::exit_codes::ExitCode;
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+use indexmap::IndexMap;
+use qiskit_accelerate::nlayout::PhysicalQubit;
+use qiskit_circuit::operations::{Operation, Param, StandardGate};
+use qiskit_transpiler::target::{InstructionProperties, Qargs, Target};
+use std::ffi::{c_char, CStr, CString};
+use std::mem::forget;
+use std::ptr::null_mut;
+
+// Helper structures
+
+/// Represent a list of instruction names coming from the target.
+#[repr(C)]
+pub struct QkTargetNameList {
+    list: *mut *mut c_char,
+    length: usize,
+}
+
+/// Represent a list of qargs coming from the target.
+#[repr(C)]
+pub struct QkTargetQargsList {
+    list: *mut QkTargetQargs,
+    length: usize,
+}
+
+/// Represent qargs coming from the target.
+#[repr(C)]
+pub struct QkTargetQargs {
+    args: *mut u32,
+    length: usize,
+}
+
+/// @ingroup QkTarget
+/// Construct a new ``Target`` with the given number of qubits.
+///
+/// @param description A string describing the ``Target``, or a NULL pointer.
+/// @param num_qubits The number of qubits the ``Target`` will explicitly support.
+///
+/// @return A pointer to the new ``Target``
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///
+/// # Safety
+///
+/// The ``description`` type is expected to be a pointer to a null-terminated valid string. If the
+/// pointer is not valid or not null ending, the behavior will be undefined.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_new(description: *mut c_char, num_qubits: usize) -> *mut Target {
```

Comment:
Me and @Cryoris were discussing about whether a description is still needed for a `Target` in the C API. Is there any use for it currently?

Reply:
After internal discussion, description is no longer part of the initial C API, removed in e5911c3